frontend/src/app/main/data/workspace/transforms.cljs

```diff
@@ -1120,7 +1132,11 @@
 
          (when (and (some? frame-id) (d/not-empty? changes))
            (rx/of (dch/commit-changes changes)
-                  (dwc/expand-collapse frame-id))))))))
+                  (dwc/expand-collapse frame-id)))
+         (when add-component-to-variant?
+           (rx/of (ptk/event ::ev/event {::ev/name "add-component-to-variant"})))
+         (when add-new-variant?
+           (rx/of (ptk/event ::ev/event {::ev/name "add-new-variant" :trigger "move-shapes-in-workspace"}))))))))
```

Comment:
The same, and the origin can be added to both events.
Also, there is a simplified event construction: `(ev/event {::ev/name ... })`